src/main/java/pro/belbix/ethparser/web3/harvest/db/HardWorkDbService.java

```diff
@@ -207,5 +210,9 @@ public void fillExtraInfo(HardWorkDTO dto) {
     if (lastHardWorkBlockDate != null) {
       dto.setIdleTime(dto.getBlockDate() - lastHardWorkBlockDate);
     }
+
+    Long lastEventBlockDate = idleTimeService.GetLastEventBlockDate(dto.getNetwork(),
```

Comment:
1) db service should work only with data base
move this logic to parser
2) we need "last doHardWork call" for other improvements
better to fetch last event by findLastLogEvent

Reply:
Moved to the Parser. But kept logic as it was before because of mandatory topics.